online-shop/src/app/services/shopping-cart.service.ts

```diff
@@ -1,12 +1,19 @@
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { AppConfig } from '../app.config';
 import { CartItem } from '../interfaces/cartItemInterface';
+import { Order } from '../interfaces/orderInterface';
 import { Product } from '../interfaces/productInterface';
 
 @Injectable({
   providedIn: 'root',
 })
 export class ShoppingCartService {
   items: CartItem[] = [];
+  private orderUrl = AppConfig.API_ENDPOINT + '/orders';
+
+  constructor(private http: HttpClient) {}
 
   addToCart(product: Product): void {
     const itemAlreadyInCart = this.items.find((item) => item.id == product.id);
```

Comment:
When comparing 2 variables the **==** ignores the datatype of variables whereas **===** is used for comparing two variables checking for the datatype as well. Thus, if you compare for example the string **"2"** with number **2**, then the **==** operator would return true, but if you compare the string **"2"** with number **2** using the **===** it would return false. In your case you want to compare 2 variables having the type **number** so you should use the **===** operator.